onnx/defs/nn/defs.cc

```diff
@@ -487,6 +487,125 @@ ONNX_OPERATOR_SCHEMA(Conv).FillUsing(ConvOpSchemaGenerator("a filter"));
 
 } // namespace ONNX_NAMESPACE
 
+void convTransposeShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // we need at least two inputs to have a shape for this inference.
+  if (!hasNInputShapes(ctx, 2)) {
+    return;
+  }
+
+  // don't bother with legacy auto_pad for now
+  if (ctx.getAttribute("auto_pad")) {
+    return;
+  }
+
+  auto input_shape = ctx.getInputType(0)->tensor_type().shape();
+  if (input_shape.dim_size() < 2) {
+    return; // The input shape is not properly set.
```

Comment:
Please change comment to "Input tensor should have at least two dimensions" ... This will be important if we convert these into error messages (as we plan to do over next few days).